opensrp-app/src/main/java/org/smartregister/sync/helper/LocationServiceHelper.java

```diff
@@ -467,22 +467,33 @@ public void fetchAllLocations(int recordCount) {
                     }.getType()
             );
 
-            for (Location location : locations) {
-                try {
-                    location.setSyncStatus(BaseRepository.TYPE_Synced);
-
-                    locationRepository.addOrUpdate(location);
-
-                    for (LocationTag tag : location.getLocationTags()) {
-                        LocationTag locationTag = new LocationTag();
-                        locationTag.setLocationId(location.getId());
-                        locationTag.setName(tag.getName());
+            if (locations != null) {
+                for (Location location : locations) {
+                    try {
+                        location.setSyncStatus(BaseRepository.TYPE_Synced);
 
-                        locationTagRepository.addOrUpdate(locationTag);
+                        locationRepository.addOrUpdate(location);
+                        if (location != null) {
```

Comment:
Is this null check necessary? Is there a possibility of any of the returned locations being null?

Reply:
Makes sense, removed it